pulsar-metadata/src/main/java/org/apache/pulsar/metadata/coordination/impl/CoordinationServiceImpl.java

```diff
@@ -80,6 +81,12 @@ public void close() throws Exception {
                 k -> new LockManagerImpl<T>(store, clazz, executor));
     }
 
+    @Override
+    public <T> LockManager<T> getLockManager(MetadataSerde<T> serde) {
+        return (LockManager<T>) lockManagers.computeIfAbsent(serde,
```

Comment:
So we require to implement equals and hashCode on MetadataSerde now?

Reply:
Oh, I didn't think of that. The thing is that the caching of the instances of `LockManager` is not very useful in practice (since we mostly call it once..). It was just there to have a collection of instances to close when we close the LockManager. In that light, the current approach could be "fine".

Reply:
Makes sense to me. Thanks